bblfsh/pyuast.cc

```diff
@@ -1,272 +1,760 @@
 #include <cstdint>
 #include <cstdlib>
 #include <cstring>
+#include <map>
 
 #include <Python.h>
+#include <structmember.h>
 
-#include "uast.h"
-#include "memtracker.h"
-
-
+#include "libuast.hpp"
 
 // Used to store references to the Pyobjects instanced in String() and
 // ItemAt() methods. Those can't be DECREF'ed to 0 because libuast uses them
 // so we pass ownership to these lists and free them at the end of filter()
-MemTracker memTracker;
 
-// WARNING: calls to Attribute MUST Py_DECREF the returned value once
-// used (or add it to the memtracker)
-static PyObject *Attribute(const void *node, const char *prop) {
-  PyObject *n = (PyObject *)node;
-  return PyObject_GetAttrString(n, prop);
+PyObject* asPyBuffer(uast::Buffer buf) {
+    PyObject* arr = PyByteArray_FromStringAndSize((const char*)(buf.ptr), buf.size);
+    free(buf.ptr);
+    return arr;
+    //return PyMemoryView_FromMemory((char*)(buf.ptr), buf.size, PyBUF_READ);
```

Comment:
Is there a reason why we want to keep this comment here?

It that is for dev documentation purpose - e.g "how to do debuging", it's cool to keep it but a comment why would really help to understand that for a future reader.